tests/test_empty_dataset.py

```diff
@@ -0,0 +1,23 @@
+import pytest
+from anomaly_detector.config import Configuration
+from anomaly_detector.exception.exceptions import EmptyDataSetNotAllowed
+from anomaly_detector.adapters.som_model_adapter import SomModelAdapter
+from anomaly_detector.adapters.som_storage_adapter import SomStorageAdapter
+import pandas
+
+CONFIGURATION_PREFIX = "LAD"
+
+
+@pytest.fixture()
+def config():
+    """Initialize configurations before testing."""
+    config = Configuration(config_yaml=".env_config.yaml")
+    return config
+
+
+def test_empty_dataset(config):
+    with pytest.raises(EmptyDataSetNotAllowed):
+        storage = SomStorageAdapter(config)
+        som = SomModelAdapter(storage_adapter=storage)
+        som.process_anomaly_score(pandas.DataFrame())
+
```

Comment:
Code isn't complete with some edge test cases. When elasticsearch doesn't have log lines our system returns an empty pandas dataframe: See code:
https://github.com/AICoE/log-anomaly-detector/blob/16e70595fcb02e650d42c5ad2455814050cb48be/anomaly_detector/storage/es_storage.py#L95-L97

Reply:
The following code doesn't catch for a empty pandas dataframe: https://github.com/AICoE/log-anomaly-detector/blob/16e70595fcb02e650d42c5ad2455814050cb48be/anomaly_detector/adapters/som_model_adapter.py#L116-L118